azurelinuxagent/agent.py

```diff
@@ -197,45 +217,58 @@ def parse_args(sys_args): # pylint: disable=R0912
     verbose = False
     debug = False
     conf_file_path = None
-    for a in sys_args: # pylint: disable=C0103
-        m = re.match("^(?:[-/]*)configuration-path:([\w/\.\-_]+)", a) # pylint: disable=W1401,C0103
+    log_collector_mode = None
+
+    for arg in sys_args: # pylint: disable=C0103
+        m = re.match("^(?:[-/]*)configuration-path:([\w/\.\-_]+)", arg) # pylint: disable=W1401,C0103
         if not m is None:
             conf_file_path = m.group(1)
             if not os.path.exists(conf_file_path):
                 print("Error: Configuration file {0} does not exist".format(
                         conf_file_path), file=sys.stderr) 
                 usage()
                 sys.exit(1)
-        
-        elif re.match("^([-/]*)deprovision\\+user", a):
+
+        m = re.match("^(?:[-/]*)-mode:([\w/\.\-_]+)", arg) # pylint: disable=W1401,C0103
+        if m is not None:
+            log_collector_mode = m.group(1)
+            if log_collector_mode != "full":
+                print("Error: Invalid value for log collector mode: {0}. Accepted value: full. "
+                      "If mode is not specified, will use normal mode.".format(log_collector_mode))
+                print(usage())
+                sys.exit(1)
```

Comment:
I'm not sure if exiting in the `parse_args` function is the right approach since its just a utility function. Wouldn't it make more sense to just raise an Exception here and let the program die from the `main` ? There's no error handling there anyways so wouldn't make a lot of difference.